primitives/sr-io/src/lib.rs

```diff
@@ -117,15 +135,21 @@ pub trait Storage {
 	/// doesn't exist at all.
 	/// If `value_out` length is smaller than the returned length, only `value_out` length bytes
 	/// are copied into `value_out`.
+	///
+	/// See `child_get` for common child api parameters.
 	fn child_read(
 		&self,
 		child_storage_key: &[u8],
+		child_definition: &[u8],
+		child_type: u32,
 		key: &[u8],
 		value_out: &mut [u8],
 		value_offset: u32,
 	) -> Option<u32> {
 		let storage_key = child_storage_key_or_panic(child_storage_key);
-		self.child_storage(storage_key, key)
+		let child_info = ChildInfo::resolve_child_info(child_type, child_definition)
+			.expect("Invalid child definition");
```

Comment:
It seems all the validation of child specification are done here, should we also check that storage_key path match the child_info variants: which means contains ':default' as there is only Default child tries for now ?

Reply:
I do not know, I did not want to merge both definition.

Reply:
currently to parameters are related, if a user do call:
(Lets say we create a new child called second_kind)
```
child::set(
  storage_key=$prefix ++ :default ++ $some_path,
  child_info=SecondKind($some_uid,
  key=b"1"
)
```
Then this call will not fail, depending on how the backend is written it can even not panic, this wrong argument will be hard to debug for the user.